File: eTouristAgencyAPI/eTouristAgencyAPI.Services/Database/Models/ReservationReview.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace eTouristAgencyAPI.Services.Database.Models;

[Table("ReservationReview")]
public partial class ReservationReview
{
    [Key]
    public Guid Id { get; set; }

    public int AccommodationRating { get; set; }

    public int ServiceRating { get; set; }

    public string Description { get; set; } = null!;

    public DateTime CreatedOn { get; set; }

    public DateTime ModifiedOn { get; set; }

    public Guid CreatedBy { get; set; }

    public Guid ModifiedBy { get; set; }

    [ForeignKey("CreatedBy")]
    [InverseProperty("ReservationReviewCreatedByNavigations")]
    public virtual User CreatedByNavigation { get; set; } = null!;

    [ForeignKey("Id")]
    [InverseProperty("ReservationReview")]
    public virtual Reservation IdNavigation { get; set; } = null!;

    [ForeignKey("ModifiedBy")]
    [InverseProperty("ReservationReviewModifiedByNavigations")]
    public virtual User ModifiedByNavigation { get; set; } = null!;
}

File: Frontend/etouristagency_mobile/lib/models/reservation_review/reservation_review.dart
class ReservationReview {
  String? id;
  int? accommodationRating;
  int? serviceRating;
  String? description;
  ReservationReviewReservationResponse? reservation;

  ReservationReview(
    this.id,
    this.accommodationRating,
    this.serviceRating,
    this.description,
    this.reservation,
  );

  factory ReservationReview.fromJson(Map<String, dynamic> json) {
    return ReservationReview(
      json["id"],
      json["accommodationRating"],
      json["serviceRating"],
      json["description"],
      json["reservation"] != null
          ? ReservationReviewReservationResponse.fromJson(json["reservation"])
          : null,
    );
  }
}

class ReservationReviewReservationResponse {
  String? userId;
  ReservationReviewUserResponse? user;

  ReservationReviewReservationResponse(this.userId, this.user);

  factory ReservationReviewReservationResponse.fromJson(
    Map<String, dynamic> json,
  ) {
    return ReservationReviewReservationResponse(
      json["userId"],
      json["user"] != null
          ? ReservationReviewUserResponse.fromJson(json["user"])
          : null,
    );
  }
}

class ReservationReviewUserResponse {
  String? id;
  String? firstName;
  String? lastName;

  ReservationReviewUserResponse(this.id, this.firstName, this.lastName);

  factory ReservationReviewUserResponse.fromJson(Map<String, dynamic> json) {
    return ReservationReviewUserResponse(
      json["id"],
      json["firstName"],
      json["lastName"],
    );
  }
}
